### DISCUSSION TITLE
Resuming PySWMM model by dynamic control policy

### DISCUSSION THREAD

**Discussion by mohsen-ghaffari1992** [pyswmm/pyswmm#436] [Q&A] [answered]
Hello everyone,

We are attempting to use PySWMM to obtain information about our model, pause the process, run a separate code to obtain a new policy, and then resume the PySWMM process with the new control policy in place. However, we have encountered several challenges in attempting to resume the PySWMM process using the new control policy. For instance, we have been unable to set the last total flow in a section, among other issues.
To make it clear, we have a scala code which generates different policies for controlling PySWMM model. In the same time, we have an script that extract different values such as water_level = J2.total_inflow - J2.total_outflow. Assume we start (time = 0, control = p0, water_level = 0), after one step we return (water_level = 10) to scala code. Then, scala code generates a new policy p1, so, we need to call python script using (time = 1, control = p1, water_level = 10), but how?!

I am hoping that you can provide guidance on whether such a process is possible and, if so, how we can effectively carry out these steps to achieve our objectives. Your expertise in this area would be greatly appreciated.
Thank you for your time and consideration. I look forward to hearing from you soon.

**Comment by bemcdonnell** [comment]
@mohsen-ghaffari1992 Sounds like a cool problem.  Yes - this is entirely possible.  This is something we do a lot.  Your might consider hosting pyswmm behind some callable API where the simulation state would remained "paused" OR you can stop the simulation and restart using a hotstart file.  The former keeps your simulation state and the latter runs pyswmm in a batch process re-initializing with the hot start.   Just so I make sure that I answer your question, are you also asking how to set the pumping rate using pyswmm?

**Reply by mohsen-ghaffari1992** [reply]
Thank you for your response @bemcdonnell. I am pleased to hear that the information you provided appears to be helpful for my problem. In regards to the hotstart file, I was wondering if you could kindly provide me with a document or resource that outlines the steps for saving a hotstart file and reopening it at a later time.
I have attempted to search for relevant documents through online resources, but have been unable to find any that offer clear guidance on this process. Your assistance in this matter would be greatly appreciated, and I look forward to hearing from you soon.
Thank you for your time and consideration.

**Reply by bemcdonnell** [reply, marked as answer]
@mohsen-ghaffari1992 

page 197 of the [SWMM users manual](https://www.epa.gov/system/files/documents/2022-04/swmm-users-manual-version-5.2.pdf) talks about hotstart files. In a [FILES] section of your input file you can specify to SAVE or/and USE a hot start file. 

![IMG_5182](https://user-images.githubusercontent.com/6105134/236293520-4041235a-e876-49f9-a9d9-446dc1fbbb5a.jpeg)
